workflows/shared.py

```diff
@@ -107,34 +113,26 @@ def free_port_selector(node_subscription_id: UUIDstr, speed: int, enum: str = "P
     return Choice(enum, zip(interfaces.keys(), interfaces.items()))  # type:ignore
 
 
-def summary_form(product_name: str, summary_data: dict) -> Generator:
-    class ProductSummary(MigrationSummary):
-        data = SummaryData(**summary_data)
-
+def summary_form(product_name: str, summary_data: SummaryData) -> Generator:
     class SummaryForm(FormPage):
-        class Config:
-            title = f"{product_name} summary"
+        model_config = ConfigDict(title=f"{product_name} summary")
 
-        product_summary: ProductSummary
+        product_summary: migration_summary(summary_data)  # type: ignore
```

Comment:
Most places in surf orchestrator we introduced a type 'MigrationSummary', something along the lines of

`ProductSummary = migration_summary(sumary_data)`

To make mypy happy this involves some casting